cql3/statements/modification_statement.cc

```diff
@@ -275,13 +276,36 @@ modification_statement::do_execute(query_processor& qp, service::query_state& qs
         return execute_with_condition(qp, qs, options);
     }
 
-    return execute_without_condition(qp, qs, options).then([] (coordinator_result<> res) {
+    return execute_without_condition(qp, qs, options).then([this, &options, &qp] (coordinator_result<> res) {
         if (!res) {
             return make_ready_future<::shared_ptr<cql_transport::messages::result_message>>(
                     seastar::make_shared<cql_transport::messages::result_message::exception>(std::move(res).assume_error()));
         }
-        return make_ready_future<::shared_ptr<cql_transport::messages::result_message>>(
+
+        json_cache_opt json_cache = maybe_prepare_json_cache(options);
+        std::vector<dht::partition_range> keys = build_partition_keys(options, json_cache);
```

Comment:
Let's avoid this double work, and prepare tablet info in get_mutations().

Reply:
The comment still stands.

Reply:
I think it show this is the wrong level. Why does modification_statement need to know about tablets? It's translating prepared CQL into storage_proxy methods.

storage_proxy will already be tablet aware. It can send back an opaque routing object to be translated to cql.


Let's pretend we enhance alternator to return routing information. Should we audit alternator code to insert routing checks? No, storage_proxy does that already and we'd just translate the returned routing results into json or an HTTP header or whatever.